extensions/blocks/opentable/use-restaurant-search.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { unionBy, throttle } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+
+export default function useRestaurantSearch( searchTerm, maxResults ) {
+	const [ restaurants, setRestaurants ] = useState( [] );
+
+	const searchRestaurants = ( input = '' ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=' +
+				maxResults +
+				'&query=' +
+				input
+		)
+			.then( result => result.json() )
+			.then( restaurantResponse =>
+				setRestaurants( unionBy( restaurants, restaurantResponse.items, 'rid' ) )
+			);
+	};
+
+	const throttledSearchRestaurants = useCallback( throttle( searchRestaurants, 500 ), [
+		restaurants,
+	] );
+
+	useEffect( () => {
+		throttledSearchRestaurants( searchTerm );
```

Comment:
When embed script is pasted in, the fetch is fired off with the embed as the query string, which causes a 403 from the OpenTable API.

<img width="625" alt="script-403" src="https://user-images.githubusercontent.com/3629020/71786177-9950cf80-306d-11ea-8058-3980825ab9dc.png">

Checking for the embed code before running the search would be one way of fixing this. 

```javascript
	useEffect( () => {
		if ( ! embedRegex.test( searchTerm ) ) {
			throttledSearchRestaurants( searchTerm );
		}
	}, [ searchTerm ] );
```

Reply:
Agreed! We should probably not fire off requests when we know that it's not a restaurant name

Reply:
I've implemented a couple of checks to prevent this from firing so much. This particular error came from the `/` characters in the embed code. I'll URL encode the search term